tests/integration_toolbox.py

```diff
@@ -155,7 +157,45 @@ def testGetUrlToFileCachedKeepaliveChanged(self):
         finally:
             conn.close()
         self.assertEqual(b'This is a test\n', data)
-
+    
+    def testGetUrlReadableWritable(self):
+        """Call get_url to trigger readable and writable"""
+        try:
+            connection = http.client.HTTPConnection("localhost:{}".format(self.port))
+            connection.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM) 
```

Comment:
What's the reason for explicitly assigning the socket and checking that it's None after? I'm confused by this pattern--HTTPConnection should automatically do all the setup that's necessary.

Reply:
This is because I wanted to hit the branch where it sets the socket to None and then returns, so I was just testing that the value of socket is correct from that branch.

Reply:
Hmm, reading that now. What needs to happen is to have an established connection here, and stale data (otherwise that isn't really hit). So instead of assigning to the socket, calling `connection.request('GET', 'localhost:{}/')` before `get_url` should do the trick--since it's starting the connection but not actually reading the data off it.

Reply:
This is causing an error from socketserver.py: `ConnectionAbortedError: [WinError 10053] An established connection was aborted by the software in your host machine` I am going to do a bit of investigation with this one

Reply:
If it winds up being a rabbit hole, we can leave that branch untested. Although it's probably worth doing a test where we just call get_url back-to-back twice so the connection is reused (it just won't do the clearing old data off the socket).

Reply:
Didn't seem to work, the weird part is that it doesn't seem like a socket is ever assigned, printing out the socket, after creating the connection, and both get_url() calls result in "None" all three times.